Classes/ContentRepository/NodeTranslationService.php

```diff
@@ -177,11 +177,16 @@ protected function translateNode(NodeInterface $sourceNode, NodeInterface $targe
         $targetNode->setHiddenInIndex($sourceNode->isHiddenInIndex());
         $targetNode->setHiddenBeforeDateTime($sourceNode->getHiddenBeforeDateTime());
         $targetNode->setHiddenAfterDateTime($sourceNode->getHiddenAfterDateTime());
+        $targetNode->setIndex($sourceNode->getIndex());
+
+        // Move node if parents are not matching
+        if ($sourceNode->getParentPath() !== $targetNode->getParentPath()) {
+            $targetNode->moveInto($sourceNode->getParent());
+        }
```

Comment:
Imho those lines here are the ones that actually belong to the bugfix that is the pr title.